gateway/src/apicast/policy/3scale_batcher/3scale_batcher.lua

```diff
@@ -110,17 +117,42 @@ local function error(service, rejection_reason)
   end
 end
 
-local function handle_backend_ok(self, transaction)
+local function update_downtime_cache(cache, transaction, backend_status, cache_handler)
+  local key = keys_helper.key_for_cached_auth(transaction)
+  cache_handler(cache, key, backend_status)
```

Comment:
Isn't this handler being called twice ? Once by APIcast policy and once by this one?

Reply:
No, because when this policy is enabled, APIcast does not authorize nor report to the 3scale backend. And the handler is only called when there's a backend response to be handled.